src/renderer/lib/hooks/useAppListeners.ts

```diff
@@ -229,7 +230,32 @@ export const useAppListeners = () => {
 
   const setDolphinOpen = useDolphinStore((store) => store.setDolphinOpen);
   ipc_dolphinClosedEvent.renderer!.useEvent(
-    async ({ dolphinType }) => {
+    async ({ dolphinType, exitCode }) => {
+      // ignore some situations
+      const ignore = [
+        exitCode === 3 && !isLinux && !isMac, // when selecting Update in game on windows, dolphin returns 3
+      ];
```

Comment:
This logic is getting complicated now with the ignoring.

Can you create a helper function with the signature `handleDolphinExitCode(exitCode: number): string | null` so that we can simply check if it's a string and emit the toast if so. The helper function returns null for the ignored situations.